src/sql-parser/src/ident.rs

```diff
@@ -64,8 +76,32 @@ macro_rules! ident {
             }
         }
 
+        #[allow(dead_code)]
+        const fn check_value(val: &str) {
+            if equal(val, ".") || equal(val, "..") {
+                panic!(stringify!(provided string literal, $val, is an invalid identifier));
+            }
+        }
+
+        const fn equal(lhs: &str, rhs: &str) -> bool {
```

Comment:
Can you leave a comment here explaining that `str` does not implement `const PartialEq`, which is why we need to hand roll this equals function?